### PR TITLE
Dashboard method moved to new service from TaskService

### PR DESCRIPTION
**Pull request by khlopkov**

### CODE REVIEW

File: ArcadiaTest/DataLayer/ITaskStatusRepository.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ArcadiaTest.DataLayer
+{
+    public interface ITaskStatusRepository
+    {
+        Task<IEnumerable<string>> FindAllTaskStatusesAsync();
```
Comment (sponkrashin) [inline ArcadiaTest/DataLayer/ITaskStatusRepository.cs]: `Find` word assumes some condition which you will use to find what you need.

File: ArcadiaTest/Controllers/TasksController.cs [hunk 2]
```diff
@@ -142,29 +146,21 @@ public IHttpActionResult Delete(int taskId)
...
         [Route("dashboard")]
-        public IHttpActionResult GetTaskDashboard()
+        public async Task<IHttpActionResult> GetTaskDashboard()
         {
             var claims = ((ClaimsIdentity)User.Identity).Claims;
             var email = claims.Where(c => c.Type == ClaimTypes.Email).First().Value;
             UserResponse currentUser;
             try
             {
                 currentUser = this._userService.GetUserWithEmail(email);
```
Comment (sponkrashin) [inline ArcadiaTest/Controllers/TasksController.cs]: Can be joined with variable declaration.

File: ArcadiaTest/BusinessLayer/StatisticsService.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        public async Task<IDictionary<string, int>> GetStatisticsOfTaskCountGroupedByStatus(int userId)
+        {
+            var result = new Dictionary<string, int>();
+
+            var statuses = await this._taskStatusRepository.FindAllTaskStatusesAsync();
+            var groupedCounts = await this._taskRepository.CountTasksGroupedByStatusAsync(userId);
+
+            foreach (var taskCount in groupedCounts)
+            {
+                result.Add(taskCount.Status, taskCount.Count);
```
Comment (sponkrashin) [inline ArcadiaTest/BusinessLayer/StatisticsService.cs]: Why do you need `result` variable?

**Comment by sponkrashin** [comment]
You should try to have only related changes in one commit. If I were you, I would have two separate commits, one for task statuses and one for statistics.

File: ArcadiaTest/DataLayer/ITaskStatusRepository.cs (hunk 1 above)
Comment (khlopkov) [reply ArcadiaTest/DataLayer/ITaskStatusRepository.cs]: https://github.com/khlopkov/ArcadiaTest2/pull/29/commits/41ef719c686f78130f8164e6acc5268daf4114c7

File: ArcadiaTest/Controllers/TasksController.cs (hunk 2 above)
Comment (khlopkov) [reply ArcadiaTest/Controllers/TasksController.cs]: https://github.com/khlopkov/ArcadiaTest2/pull/29/commits/3a267f6682a7da64e665616ad1f97d2fa901a91f

File: ArcadiaTest/BusinessLayer/StatisticsService.cs (hunk 3 above)
Comment (khlopkov) [reply ArcadiaTest/BusinessLayer/StatisticsService.cs]: https://github.com/khlopkov/ArcadiaTest2/pull/29/commits/6d1fb496416575ab8eb787e7144170f4b10da26e

File: ArcadiaTest/BusinessLayer/StatisticsService.cs [hunk 4]
```diff
@@ -20,23 +21,13 @@ ITaskStatusRepository taskStatusRepository
...
-            }
-
-            foreach (var status in statuses)
-            {
-                if (!result.ContainsKey(status))
-                    result.Add(status, 0);
-            }
-
-            return result;
+            return statuses.ToDictionary(
```
Comment (sponkrashin) [inline ArcadiaTest/BusinessLayer/StatisticsService.cs]: I meant a bit different thing, but it is also ok.

**Pull request merged**